mobius-android/src/main/java/com/spotify/mobius/android/ObservableMutableLiveData.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * -\-\-
+ * Mobius
+ * --
+ * Copyright (c) 2017-2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.mobius.android;
+
+import static com.spotify.mobius.internal_util.Preconditions.checkNotNull;
+
+import androidx.lifecycle.MutableLiveData;
+import com.spotify.mobius.EventSource;
+import com.spotify.mobius.disposables.Disposable;
+import com.spotify.mobius.functions.Consumer;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+/** An extension of MutableLiveData that allows its Active/Inactive state to be observed. */
+final class ObservableMutableLiveData<T> extends MutableLiveData<T>
+    implements EventSource<Boolean> {
+  private final List<Consumer<Boolean>> stateListeners = new ArrayList<>();
```

Comment:
i think this would do better to be a CopyOnWriteArrayList instead, and remove the synchronization blocks

Reply:
Hmm, I can convert it to that, my only concern (and why there's a copy done in `notifyListeners` is someone deciding to unsubscribe in the middle of a notify back call - so it would modify the array while its being iterated over. I'm not sure if a CopyOnWriteArrayList would fix this, would it?

Reply:
So I did experiment around with this a little, seemed to work fine but I think maybe I still need to copy the array before sending since something might remove itself from the array while iterating, and that _might_ cause issues.

Reply:
From the COWAL docs: ` the iterator is guaranteed not to throw ConcurrentModificationException. The iterator will not reflect additions, removals, or changes to the list since the iterator was created`. That class is super useful for cases like listeners/observers because it has the semantics you usually want, and it's generally better than any implementation you can come up with yourself. :)